dbms/src/Storages/Transaction/TiDB.h

```diff
@@ -398,4 +401,14 @@ String genJsonNull();
 tipb::FieldType columnInfoToFieldType(const ColumnInfo & ci);
 ColumnInfo fieldTypeToColumnInfo(const tipb::FieldType & field_type);
 
+using ColumnsWithTypeAndName = DB::ColumnsWithTypeAndName;
+using NamesAndTypes = DB::NamesAndTypes;
+using DAGColumnInfo = std::pair<String, ColumnInfo>;
+using DAGSchema = std::vector<DAGColumnInfo>;
+
+DAGSchema genSchemaFromTableScan(const tipb::TableScan & table_scan);
+NamesAndTypes genNamesAndTypesFromTableScan(const tipb::TableScan & table_scan);
+ColumnsWithTypeAndName getColumnWithTypeAndName(const DAGSchema & schema);
+ColumnsWithTypeAndName getColumnWithTypeAndName(const NamesAndTypes & names_and_types);
```

Comment:
We'd better put these methods in separate files.
In my opinion, only
`TiDBColumnInfo toTiDBColumnInfo(tipb::ColumnInfo)` fits here.

Reply:
Maybe add them in ColumnWithTypeAndName.h and NamesAndTypes.h will make sense?